ch_pipeline/hfb/containers.py

```diff
@@ -10,12 +10,40 @@
 
 from ch_util import andata
 
-from ..core.containers import RawContainer, FreqContainer
+from ..core.containers import ContainerBase, RawContainer, FreqContainer
 
 from draco.core.containers import TODContainer
 
 
-class HFBData(RawContainer, FreqContainer):
+class HFBContainer(ContainerBase):
+    """A base class for all HFB containers.
+
+    Like :class:`ContainerBase`, but with some properties specific to HFB data.
+    """
+
+    @property
+    def hfb(self) -> memh5.MemDataset:
+        """The main hfb dataset."""
+        return self.datasets["hfb"]
+
+    @property
+    def weight(self) -> memh5.MemDataset:
+        """The inverse variance weight dataset."""
+        if "weight" in self:
+            weight = self["weight"]
+        elif "hfb_weight" in self:
+            weight = self["hfb_weight"]
+        else:
+            weight = self["flags/hfb_weight"]
```

Comment:
This access will only work on subclasses of RawContainer and I believe will throw an exception as access to other groups is not allowed without `RawContainer`. You might want to protect the call a bit more.

Reply:
Done.